msvc-specific.cmake

```diff
@@ -0,0 +1,122 @@
+# Copyright (C) 2016 JRL AIST-CNRS.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#
+# This file contains a collection of macros specific to Visual Studio and the
+# MSVC compilers.
+#
+
+# GET_MSVC_VERSION
+# ----------------
+#
+# This macro return a string corresponding to the version of the MSVC compiler
+# used.
+# The value is returned in the variable MSVC_VERSION. It is an empty string if
+#  - the compiler used is not MSVC,
+#  - it is older than MSVC 6.0
+#  - or newer than MSVC 14.0
+#
+MACRO(GET_MSVC_VERSION)
+  if(MSVC60)
+    set(MSVC_VERSION "6.0")
+  elseif(MSVC70)
+    set(MSVC_VERSION "7.0")
+  elseif(MSVC71)
+    set(MSVC_VERSION "7.1")
+  elseif(MSVC80)
+    set(MSVC_VERSION "8.0")
+  elseif(MSVC90)
+    set(MSVC_VERSION "9.0")
+  elseif(MSVC10)
+    set(MSVC_VERSION "10.0")
+  elseif(MSVC11)
+    set(MSVC_VERSION "11.0")
+  elseif(MSVC12)
+    set(MSVC_VERSION "12.0")
+  elseif(MSVC14)
+    set(MSVC_VERSION "14.0")
+  else()
+    set(MSVC_VERSION "")
+	message("MSVC version not found: you're either targeting a non msvc compiler or you're using a new compiler, in which case, please update the GET_MSVC_VERSION macro")
+  endif()
+ENDMACRO(GET_MSVC_VERSION)
+
+
+# REQUEST_MINIMUM_MSVC_VERSION(VER)
+# ---------------------------------
+#
+# Return a fatal error if the current MSVC version is strictly lower than VER
+# or when no MSVC compiler or an unknown version of it is used.
+#
+# A reason for the version to be unknown is that it is newer than the versions
+# recognize by the above macro GET_MSVC_VERSION. In this case, please update 
+# the macro and its documentation.
+#
+MACRO(REQUEST_MINIMUM_MSVC_VERSION)
+  if(NOT ${ARGC} GREATER 0)
+    message("REQUEST_MINIMUM_MSVC_VERSION requires an argument" FATAL_ERROR)
+  endif()
+  GET_MSVC_VERSION()
+  if (${MSVC_VERSION})
+    if (${MSVC_VERSION} STRGREATER ${ARGV0})
+    else()
+	  message("Minimum MSVC version required is " ${ARGV0} FATAL_ERROR)
```

Comment:
Maybe it's personal but I find the numbering of Visual Studio version to be confusing (e.g. 12.0 is Visual Studio 2013, 14.0 is Visual Studio 2015 and 15.0 is Visual Studio 15) so I think it would be very helpful to report the version that was detected along with this failure.